src/context/PendingNotificationsContext.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+import { Notification, dismissNotificationAsync } from 'expo-notifications'
+
+import * as storage from '../lib/secure-storage'
+import { NotificationData } from '../types'
+
+const defaultPendingNotifications: Notification[] = []
```

Comment:
Could we extend this type to have `NotificationData` nested in order to avoid casting to it constantly?

Reply:
That sounds like a good idea, it's a pity that `Notification` is not generic itself.